x-pack/plugins/security/server/config.test.ts

```diff
@@ -2433,3 +2449,84 @@ describe('createConfig()', () => {
     });
   });
 });
+
+describe('checkFipsConfig', () => {
+  let mockExit: jest.SpyInstance;
+  beforeAll(() => {
+    mockExit = jest.spyOn(process, 'exit').mockImplementation((exitCode) => {
+      throw new Error(`Fake Exit: ${exitCode}`);
+    });
+  });
+
+  afterAll(() => {
+    mockExit.mockRestore();
+  });
+
+  it('should log an error message if FIPS mode is misconfigured - xpack.security.fipsMode.enabled true, Nodejs FIPS mode false', async () => {
+    const logger = loggingSystemMock.create().get();
+
+    try {
+      createConfig(ConfigSchema.validate({ fipsMode: { enabled: true } }), logger, {
+        isTLSEnabled: true,
+      });
+    } catch (e) {
+      expect(mockExit).toHaveBeenNthCalledWith(1, 78);
+    }
+
+    expect(loggingSystemMock.collect(logger).error).toMatchInlineSnapshot(`
+                        Array [
+                          Array [
+                            "Configuration mismatch error. xpack.security.fipsMode.enabled is set to true and the configured Node.js environment has FIPS disabled",
+                          ],
+                        ]
+                `);
+  });
+
+  it('should log an error message if FIPS mode is misconfigured - xpack.security.fipsMode.enabled false, Nodejs FIPS mode true', async () => {
+    mockGetFipsFn.mockImplementationOnce(() => {
+      return 1;
+    });
+
+    const logger = loggingSystemMock.create().get();
+
+    try {
+      createConfig(ConfigSchema.validate({ fipsMode: { enabled: false } }), logger, {
+        isTLSEnabled: true,
+      });
+    } catch (e) {
+      expect(mockExit).toHaveBeenNthCalledWith(1, 78);
+    }
+
+    expect(loggingSystemMock.collect(logger).error).toMatchInlineSnapshot(`
+                        Array [
+                          Array [
+                            "Configuration mismatch error. xpack.security.fipsMode.enabled is set to false and the configured Node.js environment has FIPS enabled",
+                          ],
+                        ]
+                `);
+  });
+
+  it('should log an info message if FIPS mode is properly configured - xpack.security.fipsMode.enabled true, Nodejs FIPS mode true', async () => {
+    mockGetFipsFn.mockImplementationOnce(() => {
+      return 1;
+    });
+
+    const logger = loggingSystemMock.create().get();
+
+    try {
+      createConfig(ConfigSchema.validate({ fipsMode: { enabled: true } }), logger, {
+        isTLSEnabled: true,
+      });
+    } catch (e) {
+      expect(mockExit).toHaveBeenNthCalledWith(1, 78);
```

Comment:
Nit: Wouldn't we expect this test case not to throw at all? It would still fail appropriately on the next check, so this doesn't change the final outcome anyway - just a nit.